src/optics.jl

```diff
@@ -130,6 +141,11 @@ const ComposedOptic{Outer,Inner} = ComposedFunction{Outer,Inner}
 outertype(::Type{ComposedOptic{Outer,Inner}}) where {Outer,Inner} = Outer
 innertype(::Type{ComposedOptic{Outer,Inner}}) where {Outer,Inner} = Inner
 
+const SALT_COMPOSEDOPTIC = make_salt(0xcf7322dcc2129a31)
+Base.hash(l::ComposedOptic, h::UInt) = hash(l.outer, hash(l.inner, SALT_COMPOSEDOPTIC + h))
+
+Base.:(==)(l::ComposedOptic, r::ComposedOptic) = l.outer == r.outer && l.inner == r.inner
```

Comment:
```suggestion
const SALT_COMPOSEDOPTIC = make_salt(0xcf7322dcc2129a31)
Base.hash(l::ComposedOptic, h::UInt) = hash(l.outer, hash(l.inner, SALT_COMPOSEDOPTIC + h))

Base.:(==)(l::ComposedOptic, r::ComposedOptic) = l.outer == r.outer && l.inner == r.inner
```
This is type piracy, but luckily I think it is also not needed.